### PR TITLE
tests: Bluetooth: shell: Fix connection reference leak

### PR DESCRIPTION
**Pull request by MariuszSkamra**
This fixes regression introduced in recently.
Redundant connection reference is taken twice in connected()
callback. The redunant reference was then not returned, and
as a result we had a leak.

### CODE REVIEW

**Review by sjanc — APPROVED** [review]

File: subsys/bluetooth/shell/bt.c [hunk 1]
```diff
@@ -2845,10 +2843,8 @@ static int cmd_connect_le(const struct shell *sh, size_t argc, char *argv[])
 					BT_GAP_SCAN_FAST_INTERVAL,
 					BT_GAP_SCAN_FAST_INTERVAL);
 
-	err = bt_conn_le_create(
-		&addr, create_params,
-		BT_LE_CONN_PARAM(BT_GAP_INIT_CONN_INT_MIN, BT_GAP_INIT_CONN_INT_MIN, 0, 400),
-		&conn);
+	err = bt_conn_le_create(&addr, create_params, BT_LE_CONN_PARAM_DEFAULT,
+				&conn);
```
Comment (jhedberg) [inline subsys/bluetooth/shell/bt.c:2847]: Seems like it'd make sense to update `bt_conn_le_create()` to accept `NULL` as the last parameter in cases like this (i.e. where you don't do anything meaningful with the pointer even after a successful return).

File: subsys/bluetooth/shell/bt.c [hunk 2]
```diff
@@ -694,8 +694,6 @@ static void connected(struct bt_conn *conn, uint8_t err)
 		}
 	}
 
-	default_conn = bt_conn_ref(conn);
```
Comment (jhedberg) [inline subsys/bluetooth/shell/bt.c:697]: I can see the bug with the ref getting taken twice for `ROLE_PERIPHERAL`, but where is `default_conn` getting set for `ROLE_CENTRAL`?

Comment (jhedberg) [reply subsys/bluetooth/shell/bt.c:697]: > I can see the bug with the ref getting taken twice for `ROLE_PERIPHERAL`, but where is `default_conn` getting set for `ROLE_CENTRAL`?

Nevermind, I'm just blind.

**Review by jhedberg — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Review by alwa-nordic — APPROVED** [review]

**Review by msmttchr — APPROVED** [review]
@MariuszSkamra, this seems to fix the double reference problem. Thanks

**Review by HoZHel — APPROVED** [review]
Also it fixes re-advertising after a disconnection.

**Pull request merged**